### PR TITLE
Feature/habit events

### PR DESCRIPTION
**Pull request by Ryden-Graham**
Got basic habit event functionality integrated with the "today's habits" page.
To interface with habits, use the methods in Account.
Habits events currently have an extra "holder" field used locally for the "today's habits" page, but unsaved on firestore
(firestore cannot save complex types, and new holders are created on load anyway if needed. Having the local variable just helps synchronize holders and events).

NOTE: Account only returns copies of its events to avoid the holders being saved to the sever. If you ever need to interface with habit events, do NOT simply redirect pointers (we want to try and restrict access to the main event lists).

Additional changes:
- Added password resets
- Added a password check to the username/pfp update page
- Removed a bunch of commented code

Many tests are currently broken; we will fix them after friends are added and there are no more core changes that could break them again.

### CODE REVIEW

**Review by hibaqw — APPROVED** [review]
Looks great!

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/ChangePasswordFragment.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                    builder.setTitle("Missing Required Field");
+                    builder.setMessage("");
+                    builder.setNegativeButton("OK", null);
+                    AlertDialog alertBox = builder.create();
+                    alertBox.show();
+                    return;
+                }
+
+                // Empty field verification
+                if (oldPassword.equals(newPassword)) {
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/ChangePasswordFragment.java:103]: wrong comment

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/UserProfilePage.java [hunk 2]
```diff
@@ -65,13 +69,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        passwordVerified = false;
+        new VerifyPasswordFragment().show(getSupportFragmentManager(),"VERIFY_PASSWORD");
         // create a new settings fragment and display it on the appropriate frame
-        Fragment userSettings = new UserSettingsFragment();
-        // begin fragment transaction and add current activity to backstack
-        transaction = manager.beginTransaction();
-        transaction.add(R.id.settingsFrame, userSettings);
-        transaction.addToBackStack(null);
-        transaction.commit();
+        if (passwordVerified) {
```
Comment (ccchoi2) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/UserProfilePage.java:75]: remove this

**Review by adiep2 — CHANGES_REQUESTED** [review]
There is a bug when editing the habits causing it to not show up on the daily habit page.  If you create a habit with no other habits in the list then edit the frequency to include today it will not update the daily habit page.
Steps to reproduce:
1. Clear all habits
2. Add a habit without today in the frequency
3. Edit habit to include today in the frequency
4. Habit does not show up in daily habits page

**Review by calchoi028 — APPROVED** [review]
LGTM

**Review by adiep2 — APPROVED** [review]
Looks good!

**Pull request merged**